src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3591,6 +3607,15 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   case _ => tp
                 }
 
+                // Inline RHS of ValDef for right-associative by-value operator desugaring
+                val args2 = (args1, mt.params) match {
+                  case ((ident: Ident) :: Nil, param :: Nil) if param.isByNameParam && rightAssocValDefs.contains(ident.symbol) =>
+                    inlinedRightAssocValDefs += ident.symbol
+                    val rhs = rightAssocValDefs(ident.symbol)
```

Comment:
> I'd prefer a symbol attachment over the two collections.

Are you sure? I checked for other uses of attachments and they all seem to be for communication between phases. I didn't find any precedent for data of a single phase being stored in attachments.

> skip the test in typedValDef - what value does it add?

It stores the RHS which is needed in doTypedApply. Is there a better way to get it?